### DISCUSSION TITLE
Need help launching nicotine-plus headless mode in termux

### DISCUSSION THREAD

**Discussion by GUUser91** [nicotine-plus/nicotine-plus#2263] [Q&A] [answered]
I need help launching nicotine-plus headless mode in the Android app termux
I get this error message
`nicotine -n`
`Traceback (most recent call last):                                                        File "/data/data/com.termux/files/usr/bin/nicotine", line 38, in <module>
    sys.exit(load_module())                                                                          ^^^^^^^^^^^^^
  File "/data/data/com.termux/files/usr/bin/nicotine", line 33, in load_module              from pynicotine import run
  File "/data/data/com.termux/files/usr/lib/python3.11/site-packages/pynicotine/__init__.py", line 204, in <module>
    apply_translations()                                                                  File "/data/data/com.termux/files/usr/lib/python3.11/site-packages/pynicotine/i18n.py", line 121, in apply_translations
    locale.bindtextdomain(TRANSLATION_DOMAIN, mo_path)
OSError: [Errno 22] Invalid argument`
The mo files are located here
/data/data/com.termux/files/usr/share/locale/uk/LC_MESSAGES/nicotine.mo

**Comment by slook** [comment]
Nicotine+ cannot be run on Android (even though it theory, headless mode could be possible - contributions/testing would be welcome).

"Seeker" is the Android client for the Soulseek peer-to-peer network, see https://github.com/jackBonadies/SeekerAndroid

**Reply by mathiascode** [reply]
Looking at the output, headless mode is almost working in Termux. There's just some incompatibility related to the locale module that needs to be addressed.

**Reply by GUUser91** [reply]
@slook
Sharing via the seeker app can only last 1-3 days before it shuts down even with battery usage set to unrestricted

**Reply by slook** [reply]
Do report the Seeker issue on https://github.com/jackBonadies/SeekerAndroid/issues to @jackBonadies will be interested to improve that.

Please test latest master branch of Nicotine+ 3.3.0.dev3 to see if the recent commit fixed the issue for you https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md

**Reply by GUUser91** [reply]
@slook
Thanks for the quick update! That update fixed the problem. You're a lifesaver!

**Reply by slook** [reply]
You're welcome! Thanks for your testing. In the near future some CLI command functions will be added so please stay up to date with latest master and feel free to report any issues you encounter.

**Reply by slook** [reply]
@GUUser91 Most of the basic core commands have now been ported into the headless CLI in Nicotine+ 3.3.0.dev4

https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md

Type `help` for a list of available commands:

```
help
Listing 12 available commands:

Nicotine+ Commands:
	/help, /? [query]  -  List available commands
	/away, /a  -  Toggle away status
	/quit, /q, /exit [-force]  -  Quit Nicotine+
	/sample, /demo <choice1|choice2> <something..>  -  Sample command description

Users:
	/add, /buddy <user>  -  Add user to buddy list
	/rem, /unbuddy <buddy>  -  Remove buddy from buddy list

Network Filters:
	/ip <user or ip>  -  Show IP address or username
	/ban <user or ip>  -  Block connections from user or IP address
	/unban <user or ip>  -  Remove user or IP address from ban lists

Configure Shares:
	/rescan [-force]  -  Rescan shares
	/shares, /ls [public] [buddy]  -  List shares

Plugin Commands:
	/plugin <toggle|info> <plugin_name>  -  Manage plugin

Type /help [query] to list similar commands
```

More commands will eventually be added.

**Comment by GUUser91** [comment, marked as answer]
Here is a how to guide for those struggling to install nicotine+ on termux. First install these dependencies
`pkg install libcairo xorgproto gettext git python`
Then install pycairo
`pip install pycairo`
Now install nicotine plus
`pip install --upgrade git+https://github.com/nicotine-plus/nicotine-plus.git`
